components/switch/switch-visibility.js

```diff
@@ -88,7 +88,10 @@ class VisibilitySwitch extends LocalizeCoreElement(SwitchMixin(LitElement)) {
 	}
 
 	_handleConditionsSlotChange(e) {
-		this._hasConditions = e.target.assignedNodes({ flatten: true }).length > 0;
+		const nodes = e.target.assignedNodes({ flatten: true });
+		if (nodes[0]?.textContent?.trim()) {
```

Comment:
Is there a scenario where `_hasConditions` would not get updated when it should? The code kinda assumes that it would be in its initial state of `false`. Perhaps this should be:
```javascript
this._hasConditions = (nodes[0]?.textContent?.trim() && nodes.length > 0);
```

Reply:
We do set the initial value to `false` in the constructor, but you are right, we would want the bool to update in cases where, for example, we initially had slot contents, but then, for whatever reason, they were deleted.

Though, would we even need the `nodes.length > 0` at that point?

Reply:
Probably not - I think that's already covered by the first check.